ui/src/app/applications/components/utils.tsx

```diff
@@ -1226,14 +1275,42 @@ export const urlPattern = new RegExp(
     )
 );
 
-export function appQualifiedName(app: appModels.Application, nsEnabled: boolean): string {
+export function appQualifiedName(app: appModels.AbstractApplication, nsEnabled: boolean): string {
     return (nsEnabled ? app.metadata.namespace + '/' : '') + app.metadata.name;
 }
 
-export function appInstanceName(app: appModels.Application): string {
+export function appInstanceName(app: appModels.AbstractApplication): string {
     return app.metadata.namespace + '_' + app.metadata.name;
 }
 
+export function isApp(abstractApp: appModels.AbstractApplication): abstractApp is appModels.Application {
+    return abstractApp.kind === "Application";
+}
+
+export function isInvokedFromApps(): boolean {
+    // console.log("*** ctx.history.location.pathname *** " + ctx.history.location.pathname)
+    const isInvokedFromApp = false //ctx.history.location.pathname.includes("applicationsets") ? true : false
```

Comment:
This is problematic. I'm trying to figure out a solution.

Reply:
Thanks a lot!
Does the rest look OK, under the listed limitations?